### PR TITLE
feat(config)!: prefer active python by default

### PR DESCRIPTION
**Pull request by finswimmer**
From now on Poetry prefers the currently active Python by default. To restore the old behavior the option `virtualenvs.use-poetry-python` was introduced and must be set to `true` (default: `false`).

`virtualenvs.prefer-active-python` is removed as an option.

### CODE REVIEW

**Comment by Secrus** [comment]
I was actually considering if we could remove that option or invert it so that one can choose to use system Python instead of the active one. That seems more instinctive behaviour than what we have currently

**Comment by finswimmer** [comment]
> I was actually considering if we could remove that option or invert it 

I wouldn't remove it, but inverting the default would be fine (But I'm biased here, because that's the config I change first in an new installation ;)

**Comment by Secrus** [comment]
Well, we could take advantage of the fact that we are making a breaking release and change that to `use-system-python` that would force usage of the Python version Poetry was installed with. That would give us an open interface to introduce `findpython` backend without breaking changes but as an alternative (until deemed stable).

**Comment by finswimmer** [comment]
I'm fine with renaming, but I think `use-system-python` is confusing for the users. While it is consistent with the code base, I guess most users would expect that the Python of their system is meant. Maybe `use-application-python` or `use-poetry-python`?

**Comment by Secrus** [comment]
> I'm fine with renaming, but I think `use-system-python` is confusing for the users. While it is consistent with the code base, I guess most users would expect that the Python of their system is meant. Maybe `use-application-python` or `use-poetry-python`?

I am fine with `use-poetry-python` and a proper docs entry explaining that.

**Review by Secrus — DISMISSED** [review]

File: docs/configuration.md [hunk 1]
```diff
@@ -445,6 +432,19 @@ If set to `false`, Python version used during Poetry installation is used.
...
+**Type**: `boolean`
+
+**Default**: `false`
+
+**Environment Variable**: `POETRY_VIRTUALENVS_USE_POETRY_PYTHON`
+
+*Introduced in 2.0.0*
+
+By default, Poetry will use the activated Python version to create a new virtual environment.
+If set to `true`, Python version used during Poetry installation is used.
```
Comment (radoering) [inline docs/configuration.md:446]:
```suggestion
If set to `true`, the Python version used during Poetry installation is used.
```

**Review by Secrus — DISMISSED** [review]

**Review by radoering — APPROVED** [review]

**Pull request merged**